src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -796,6 +808,33 @@ void Transformations::MainSnippets(void) {
     };
 #endif // OPENVINO_ARCH_X86_64
 
+    auto has_supported_in_out = [](const std::shared_ptr<const ov::Node> &n) -> bool {
+        auto supported = [&n](descriptor::Tensor& t) -> bool {
+            auto get_supported_element_types = []() {
+#if defined(OPENVINO_ARCH_ARM64)
+                static const std::set<ov::element::Type> supported_element_types =
+                    { ov::element::f32 };
+#else
+                static const std::set<ov::element::Type> supported_element_types =
+                    { ov::element::f32, ov::element::bf16, ov::element::i8, ov::element::u8 };
+#endif
+                return supported_element_types;
+            };
+            // TODO [122585] Need to add dynamic rank support
+            if (t.get_partial_shape().rank().is_dynamic())
+                return false;
+            // TODO [105804] int32 isn't supported in general because i32 emitters are required for bit-exact i32 calculations in some cases
+            //  So i32 is supported exclusively for transposes and broadcast
+            return get_supported_element_types().count(t.get_element_type()) != 0 ||
```

Comment:
Seems like you just copied this condition from `collapse_subgraph.cpp` to filter precisions...

What's about to remove precision check from Snippets library and move to plugin callbacks?
Snippets library supports different precisions since we have `PropagatePrecision` pass. 
Let the plugin decide which data types it supports using callbacks.

@IvanNovoselov what do you think? I suggest to remove precision [check](https://github.com/openvinotoolkit/openvino/blob/da7018a7dc13de67a521a4a0ff071c0781e94831/src/common/snippets/src/pass/collapse_subgraph.cpp#L197) from tokenization pass and move it to plugin callbacks.

Reply:
@a-sidorova, I agree let's do it

Reply:
Applied. Thanks Alexandra!